### ISSUE TITLE
mix appsignal.install

### ISSUE THREAD

**Issue by jeffkreeftmeijer** [appsignal/appsignal-elixir#45]
## WIZARD STEP 0

1. Ruby [old Ruby flow]
2. Elixir [new Elixir flow]

## WIZARD STEP 1

1. Mix.exs
2. Appsignal started before application
3. Mix deps.get

## CLI STEP 1

1. Generate Config file
    - receive push_api_key
    - get app name
2. Environment variables
3. PUT how to do it

## CLI STEP 2

1. Check environments
2. Add config per environment
3. Exclude test

## CLI STEP 3

1. AppSignal DEMO

## CLI STEP 4

Using Phoenix? 
  - HTTP requests
  - Template
  - Queries
  - Instrumentation hooks
 Check docs at ...

## WIZARD 3

Using Phoenix? 
- HTTP requests
- Template
- Queries
- Instrumentation hooks

Check docs at ..

## TODO

- [x] Send demo samples on installation completion #136 
- [x] Link to docs on Phoenix detection

**Comment by tombruijn** [comment]
So I'm looking at ways to automate this. I don't have enough indepth knowledge of the internals of Phoenix to know if we can shorten the steps in this phoenix integration guide (http://docs.appsignal.com/elixir/integrations/phoenix.html) by somehow configuring Phoenix application for the application in the appsignal package.

Assuming we can't automate it further for phoenix inside the AppSignal Elixir package, this is what I think this tool will become:

## mix appsignal.install push_api_key

### Pure Elixir applications

- get `push_api_key` from CLI command argument
- ask for application name with prompt
- Open `config/config.exs` file and add the following:

```elixir
# config/config.exs
config :appsignal, :config,
  name: :my_first_app,
  push_api_key: "your-hex-appsignal-key"
```

- Optional: Ask for environments to activate: dev, test, prod and inject the following in the environment files that shouldn't be activated. See also #86, if we take that approach we should inject `active: true` in the ones that should be active

```elixir
# config/dev.exs
config :appsignal, :config,
  active: false
```

- Tell the user to visit the documentation for further steps

### Phoenix applications

- Same steps as above for "Pure Elixir applications"
- following this guide: http://docs.appsignal.com/elixir/integrations/phoenix.html
  - automatically inject lines of configuration that we can automate
- Tell the user to visit the documentation for further steps

## Questions

- Can we pass along the push_api_key like so? `mix appsignal.install push_api_key` -- implementation detail
- Does the above make sense?
- I think automatically injecting code in so many places for Phoenix will become quite fragile. It could break easily if anything is non-standard or Phoenix changes anything. Which makes me worry if this approach would really work. Do we really want to automate if it can break easily? Or should we ask users to follow a guide instead? WDYT @jeffkreeftmeijer @arjan ?

**Comment by tombruijn** [comment]
Closed by #125 and #136
